### PR TITLE
[Master] Add tests for XML template expression in a conditional expression

### LINKED ISSUE #32959: Cannot use an XML template expression in a conditional expression (elvis) that is an operand of an additive expression
**Description:**
$title.

**Steps to reproduce:**
```ballerina
public function main() {
    xml? s = xml `foo`;
    xml t = (s ?: xml `<baz/>`); // OK
    xml u = (s is xml ? s : xml `<baz/>`) + xml ``; // OK

    xml v = (s ?: xml `<baz/>`) + xml ``; // error: expected 'xml', found 'xml:Element' for xml `<baz/>`
}
```

**Affected Versions:**
slbeta3-rc3

### PR DESCRIPTION
**Pull request by prakanth97**
## Purpose
Add tests for solved issue.

Fixes #32959

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml-literals.bal [hunk 1]
```diff
@@ -402,3 +402,17 @@ function testQueryInXMLTemplateExpr() {
...
+
+    xml x1 = (s ?: xml `<baz/>`);
+    test:assertEquals(x1.toString(), target);
+
+    xml x2 = (s is xml ? s : xml `<baz/>`) + xml ``;
+    test:assertEquals(x2.toString(), target);
+
+    xml x3 = (s ?: xml `<baz/>`) + xml ``;
+    test:assertEquals(x3.toString(), target);
+}
```
Comment (lasinicl) [inline tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml-literals.bal:418]: Shall we add some more tests with subtypes of `xml`?
Ex:
```ballerina
type XMLType xml:Comment|xml:ProcessingInstruction;

public function main() {
    xml? s = ();
    xml:Element e1 = xml `<baz/>`; 
    xml v1 = (s ?: e1) + xml `<a>Some Element</a>`; 

    xml<xml:Element> e2 = xml `<baz/>`;
    xml v2 = (s ?: e2) + xml `<a>Some Element</a>`; 

    XMLType e3 = xml `<!--This is a comment-->`;
    xml v3 = (s ?: e3) + xml `<a>Some Element</a>`;
}
```

Comment (prakanth97) [reply tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml-literals.bal:438]: Updated 486b106f2c018bd3a60e5d5ac5c774bd5ab8e3df

**Review by lasinicl — APPROVED** [review]

File: tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml-literals.bal [hunk 2]
```diff
@@ -402,3 +402,37 @@ function testQueryInXMLTemplateExpr() {
...
+    xml v2 = (s ?: e2) + xml `<element>A</element>`;
+    test:assertEquals(v2.toString(), "<baz></baz><element>A</element>");
+
+    XMLType e3 = xml `<!--This is a comment-->`;
+    xml v3 = (s ?: e3) + xml `<element>A</element>`;
+    test:assertEquals(v3.toString(), "<!--This is a comment--><element>A</element>");
+
+    XMLType e4 = xml `<?target data?>`;
+    xml v4 = (s ?: e4) + xml `<element>A</element>`;
+    test:assertEquals(v4.toString(), "<?target data?><element>A</element>");
```
Comment (pcnfernando) [inline tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml-literals.bal:437]: Can you add a few tests with nested elvis and ternary

Comment (prakanth97) [reply tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml-literals.bal:437]: I will update

Comment (prakanth97) [reply tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml-literals.bal:437]: Updated b10f0c513cdd5668d3e970c908263e156daff801

**Review by lasinicl — APPROVED** [review]

**Review by pcnfernando — APPROVED** [review]

**Pull request merged**